### PR TITLE
protect from having a nil string

### PR DESCRIPTION
**Pull request by pvinis**
The type of this PR is: **Bugfix**

This PR resolves [MX-]

### Description

`domain` can get `nil` as a value and that crashes app in the line below with `stringByAppendingString`.

We add a check and return a silly url to make the app not crash. This happens when the locally saved echo.json is broken, something like `{"error": "forbidden"}`. This shouldn't happen ever and we will address this in the next PR. In this PR we are trying to fix the crashing of the app in the case that the broken echo.json got saved, and instead of crashing forever, we just can't access auctions?

Are we thinking correctly here or do we need to do something else?

#trivial 

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))

[MX-434]: https://artsyproduct.atlassian.net/browse/MX-434

### CODE REVIEW

File: Artsy/App/ARSwitchBoard+Eigen.m [hunk 1]
```diff
@@ -34,6 +34,9 @@ -(NSURL *)liveAuctionsURL
     BOOL useStaging = [AROptions boolForOption:ARUseStagingDefault];
     NSString *echoDomainKey = useStaging ? @"ARLiveAuctionsStagingURLDomain" : @"ARLiveAuctionsURLDomain";
     NSString *domain = self.echo.routes[echoDomainKey].path;
+    if (domain == nil) {
+        return [NSURL URLWithString:@"https://notready.yet"];
```
Comment (ds300) [inline Artsy/App/ARSwitchBoard+Eigen.m:38]: might as well use the actual values here

```suggestion
        return [NSURL URLWithString:(useStaging ? @"live-staging.artsy.net" : @"live.artsy.net")];
```

Comment (ds300) [inline Artsy/App/ARSwitchBoard+Eigen.m:38]: might as well use the actual values here 

```suggestion
        return [NSURL URLWithString:(useStaging ? @"https://live-staging.artsy.net" : @"https://live.artsy.net")];
```

Comment (pvinis) [reply Artsy/App/ARSwitchBoard+Eigen.m]: Makes sense. No harm in duplicating these here.

**Review by ashfurrow — APPROVED** [review]
Looks good 👍

**Pull request merged**